docs/examples/classification.ipynb

```diff
@@ -0,0 +1,946 @@
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>split</th>\n",
+       "      <th></th>\n",
+       "      <th></th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>test</th>\n",
+       "      <td>0.673457</td>\n",
+       "      <td>0.326543</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>train</th>\n",
+       "      <td>0.669891</td>\n",
+       "      <td>0.330109</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "binary_response     False     True \n",
+       "split                              \n",
+       "test             0.673457  0.326543\n",
+       "train            0.669891  0.330109"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "count_by_split = df.groupby(\"split\").apply(len)\n",
+    "grouper = df.groupby([\"split\", y_var])\n",
+    "\n",
+    "(df.groupby([\"split\", y_var]).apply(lambda x: x['count'].sum())\n",
+    " / count_by_split).unstack(y_var)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2. Models\n",
+    "We aim at predicting the conditional expectation $E(Y|X)$ and will finally evaluate the models with the log loss $$S(y, z) = -ylog(z) - (1 - y) log (1 - z)$$\n",
+    "### 2.1 The trivial model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.dummy import DummyClassifier\n",
+    "\n",
+    "\n",
+    "m_trivial = DummyClassifier(strategy=\"prior\").fit(X_train, y_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2.2 Logistic\n",
+    "\n",
+    "Here, we train a Logistic Regression model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.compose import ColumnTransformer\n",
+    "from sklearn.preprocessing import OneHotEncoder\n",
+    "from sklearn.pipeline import Pipeline\n",
+    "from sklearn.linear_model import LogisticRegression\n",
+    "\n",
+    "\n",
+    "x_discrete = ['parents', 'has_nurs', 'form', 'children', 'housing', 'finance',\n",
+    "              'social', 'health']\n",
+    "# ColumnTransformer for linear models\n",
+    "col_trans_cat = ColumnTransformer(\n",
+    "    [\n",
+    "        (\n",
+    "            \"categorical_features\",\n",
+    "            OneHotEncoder(drop=\"first\", sparse_output=False),\n",
+    "            x_discrete\n",
+    "        ),\n",
+    "    ]\n",
+    ")\n",
+    "m_logistic = Pipeline(\n",
+    "    [\n",
+    "        (\"column_transformer\", col_trans_cat),\n",
+    "        (\n",
+    "            \"model\",\n",
+    "            LogisticRegression(max_iter=10000\n",
```

Comment:
I would set the penalty to zero, i.e. `penalty=None` and use `solver="newton-cholesky"`, then you can (veeeeery likely) remove the `max_iter` argument instead.

Reply:
done, also available in 8d3e7206d7ee6c0c4ab1fa004bdf4dbf33e7a346